sdk/appconfiguration/azappconfig/client.go

```diff
@@ -0,0 +1,443 @@
+func (c *Client) UpdateSyncToken(token string) {
+	c.syncTokenPolicy.addToken(token)
+}
+
+func (cs Setting) toGeneratedPutOptions(ifMatch *azcore.ETag, ifNoneMatch *azcore.ETag) *generated.AzureAppConfigurationClientPutKeyValueOptions {
+	return &generated.AzureAppConfigurationClientPutKeyValueOptions{
+		Entity:      cs.toGenerated(),
+		IfMatch:     (*string)(ifMatch),
+		IfNoneMatch: (*string)(ifNoneMatch),
+		Label:       cs.Label,
+	}
+}
+
+// AddSettingResult contains the response from AddSetting method.
+type AddSettingResult struct {
+	Setting
+
+	// Sync token for the Azure App Configuration client, corresponding to the current state of the client.
+	SyncToken *string
+}
+
+func fromGeneratedAdd(g generated.AzureAppConfigurationClientPutKeyValueResponse) AddSettingResult {
+	return AddSettingResult{
+		Setting:   settingFromGenerated(g.KeyValue),
+		SyncToken: g.SyncToken,
+	}
+}
+
+// AddSettingOptions contains the optional parameters for the AddSetting method.
+type AddSettingOptions struct {
+}
+
+// AddSetting creates a configuration setting only if the setting does not already exist in the configuration store.
+func (c *Client) AddSetting(ctx context.Context, setting Setting, options *AddSettingOptions) (AddSettingResult, error) {
+	etagAny := azcore.ETagAny
+	resp, err := c.appConfigClient.PutKeyValue(ctx, *setting.Key, setting.toGeneratedPutOptions(nil, &etagAny))
+	if err != nil {
+		return AddSettingResult{}, err
+	}
+
+	return (AddSettingResult)(fromGeneratedAdd(resp)), nil
+}
+
+// DeleteSettingResult contains the response from DeleteSetting method.
+type DeleteSettingResult struct {
+	Setting
+
+	// Sync token for the Azure App Configuration client, corresponding to the current state of the client.
+	SyncToken *string
+}
+
+func fromGeneratedDelete(g generated.AzureAppConfigurationClientDeleteKeyValueResponse) DeleteSettingResult {
+	return DeleteSettingResult{
+		Setting:   settingFromGenerated(g.KeyValue),
+		SyncToken: g.SyncToken,
+	}
+}
+
+// DeleteSettingOptions contains the optional parameters for the DeleteSetting method.
+type DeleteSettingOptions struct {
+	// If set to true and the configuration setting exists in the configuration store,
+	// delete the setting if the passed-in configuration setting is the same version as the one in the configuration store.
+	// The setting versions are the same if their ETag fields match.
+	OnlyIfUnchanged bool
+}
+
+func (cs Setting) toGeneratedDeleteOptions(ifMatch *azcore.ETag) *generated.AzureAppConfigurationClientDeleteKeyValueOptions {
+	return &generated.AzureAppConfigurationClientDeleteKeyValueOptions{
+		IfMatch: (*string)(ifMatch),
+		Label:   cs.Label,
+	}
+}
+
+// DeleteSetting deletes a configuration setting from the configuration store.
+func (c *Client) DeleteSetting(ctx context.Context, setting Setting, options *DeleteSettingOptions) (DeleteSettingResult, error) {
```

Comment:
Why are we returning `***Result`? I thought the guideline was `***Response`, I'm assuming Jeff okayed this but curious why the difference.

Reply:
Good point.  The pattern is to return `FooResponse`.

Reply:
Jeff told me that there will be a new way of getting raw responses, it is no longer needed in methods.

Reply:
The change we are referring to is `DeleteSettingResult` should be `DeleteSettingResponse`